### ISSUE TITLE
Possible unreachable code in segment.go

### ISSUE THREAD

**Issue by jxofficial** [travisjeffery/proglog#9]
https://github.com/travisjeffery/proglog/blob/f6dae5fb8cb4058d7785adffab864133d9575053/WriteALogPackage/internal/log/segment.go#L145

Given that `j` and `k` are `uint64`, the condition `j >= 0` will always be satisfied. 

Furthermore, based on this logic, _it seems to be assumed_ that `k` is non-zero, as division by 0 should cause an error. Might be something worth mentioning.

**Comment by cschar** [comment]
Yes, Just came across this code segment. the parameter j is of type uint64. bottom return will never be reached.
